This patch makes various tests pass on Solaris. There are two differences

 a) the test expects that 'tail a b c' displays tail for every file and
prepends the file name. tail(1) does not do that, we have to use gtail(1)

    15273738 tail should behave like head (multiple files)

 b) /usr/bin/false on Solaris returns 255 while on Linux 1

Prepending /usr/gnu/bin into $PATH does not fix the problems, so the next best
thing is to patch the expected results.

This is Solaris specific and it is unlikely to be accepted by upstream

--- mercurial-5.1.1/tests/test-merge1.t
+++ mercurial-5.1.1/tests/test-merge1.t
@@ -31,7 +31,7 @@ of the files in a commit we're updating
   $ mkdir b && touch b/nonempty
   $ hg up
   abort: Unlinking directory not permitted: *$TESTTMP/t/b* (glob) (windows !)
-  abort: Directory not empty: '?\$TESTTMP/t/b'? (re) (no-windows !)
+  abort: File exists: '$TESTTMP/t/b'
   [255]
   $ hg ci
   abort: last update was interrupted
--- mercurial-5.1.1/tests/test-narrow-expanddirstate.t
+++ mercurial-5.1.1/tests/test-narrow-expanddirstate.t
@@ -103,9 +103,9 @@ > EOF
 Since we do not have the ability to rely on a virtual filesystem or
 remotefilelog in the test, we just fake it by copying the data from the 'master'
 repo.
-  $ cp -a ../master/.hg/store/data/* .hg/store/data
+  $ cp -dR ../master/.hg/store/data/* .hg/store/data
 Do that for patchdir as well.
-  $ cp -a ../master/patchdir .
+  $ cp -dR ../master/patchdir .
 
 `hg status` will now add outside, but not patchdir.
   $ DIRSTATEINCLUDES=path:outside hg status
--- mercurial-5.1.1/tests/test-update-names.t
+++ mercurial-5.1.1/tests/test-update-names.t
@@ -51,7 +51,7 @@ $ hg st
   ? name/file
   $ hg up 1
   abort: Unlinking directory not permitted: *$TESTTMP/r1/r2/name* (glob) (windows !)
-  abort: Directory not empty: '?\$TESTTMP/r1/r2/name'? (re) (no-windows !)
+  abort: File exists: '$TESTTMP/r1/r2/name'
   [255]
   $ cat name/file
   text
